src/main/java/bowling/domain/Players.java

```diff
@@ -0,0 +1,54 @@
+package bowling.domain;
+
+import bowling.dto.PlayResult;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Players {
+    private final List<Player> players;
+
+    public Players(List<Player> players) {
+        validateSize(players);
+        this.players = players;
+    }
+
+    private void validateSize(List<Player> players) {
+        if (players.isEmpty()) {
+            throw new IllegalArgumentException("player수가 너무 적습니다.");
+        }
+    }
+
+    public Players(List<String> names, int totalNumberOfFrame) {
+        this(names.stream()
+                .map(name -> new Player(name, totalNumberOfFrame))
+                .collect(Collectors.toList()));
+    }
+
+    public void play(int pintCount) {
+        findLeastPlayedPlayer().addPinCount(pintCount);
+    }
+
+    public boolean isDone() {
+        return players.stream()
+                .allMatch(Player::isDone);
+    }
+
+    public Player playingPlayer() {
+        return findLeastPlayedPlayer();
+    }
+
+    private Player findLeastPlayedPlayer() {
+        return players.stream()
+                .filter(player -> !player.isDone())
+                .min(Comparator.comparing(Player::currentFrameNumber))
+                .orElseThrow(() -> new IllegalStateException("플레이어가 없습니다."));
+    }
+
+    public List<PlayResult> playResults() {
+        return players.stream()
+                .map(Player::playResult)
+                .collect(Collectors.toList());
```

Comment:
외부에 List를 반환할 경우 Read-only로 반환하여 외부에서 컬렉션을 변경할 수 없도록 하시면 더 좋을 것 같아요 :)
```suggestion
                .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
```

Reply:
이 부분도 반영하겠습니다. 👍